api/calculate_frequencies.py

```diff
@@ -117,19 +64,85 @@ def calculate_frequencies3(genotypedata, alleles_definitions):
 
 	ncol = max(freq_length)
 
+	# create matrix with frequency values from frequencies list
+	freqmatrix = _create_frequencyMatrix(nloci, ncol, frequencies)
+
+	# final result
+	ret = _pack_result(freq_length, freqmatrix, variability)
+
+	return ret
+
+def _getLocinames(genotypedata: pd.DataFrame):
+	col_names = list(genotypedata.columns.values)[1:]
+	prev_pos = None
+	locinames = {}
+	lociname_index = 0
+	lociname_end_index = 0
+	for name in col_names:
+		res = name.split("_")
+		if prev_pos == None:
+			prev_pos = res[0]
+		elif prev_pos != res[0]:
+			locinames[lociname_index] = (prev_pos, lociname_end_index)
+			prev_pos = res[0]
+			lociname_index += 1
+			lociname_end_index += 1
+		else:
+			lociname_end_index += 1
+			if (lociname_end_index == len(col_names)-1):
+				locinames[lociname_index] = (prev_pos, lociname_end_index)
+	return locinames
+
+def _getRawAlleles(genotypedata: pd.DataFrame, n: int, last_index: int):
+	raw_alleles = []
+	while (n <= last_index):
+		raw_alleles += genotypedata.iloc[:, n+1].tolist()
+		n += 1
+	raw_alleles = [loci for loci in raw_alleles if str(loci) != 'nan']
+	return raw_alleles, n
+
+def _get_sumList(nrows: int, raw_alleles: list, low: pd.core.series.Series, high: pd.core.series.Series):
+	sum_list = [] # needed for storing frequency values
+	sd_list = [] # standard deviation
+	for i in range(nrows):
+		tf_table = []
+		sum = 0
+		for allele in raw_alleles:
+			eval = allele > low[i] and allele <= high[i]
+			tf_table.append(eval)
+			if eval:
+				sum += 1
+		sum_list.append(sum)
+
+		true_items = []
+		for eval_i in range(len(tf_table)):
+			if tf_table[eval_i]:
+				true_items.append(raw_alleles[eval_i])
+
+		if len(true_items) > 1:
+			sd_list.append(statistics.stdev(true_items))
+	sum_list = np.array(sum_list)
+
+	# mean of standard deviation
+	meanSD = 0
+	if (len(sd_list) > 0):
+		meanSD = np.mean(sd_list)
+
+	return sum_list, meanSD
+
+def _create_frequencyMatrix(nloci: int, ncol: np.int64, frequencies: list):
 	# initialize frequency matrix with zeros
 	freqmatrix = np.zeros([nloci, ncol])
 
 	# fill out each box in the frequency matrix with frequency values from frequencies list
 	for j in range(nloci):
 		for i in range(len(frequencies[j])):
 			freqmatrix[j][i] = frequencies[j][i]
+	return freqmatrix
 
-	# ret = list
-	# add freq_length, frequency matrix, and variability
-	# ret needed to be list type to add each numpy array and matrix
+def _pack_result(freq_length: np.ndarray, freqmatrix: np.ndarray, variability: np.ndarray):
```

Comment:
This'll lead to changes in other files, but: instead of putting all of these in a list, could you instead put them in a [named tuple](https://pymotw.com/2/collections/namedtuple.html), or a small class w/ each array as a public variable?

Since these 3 matrices don't really fit together, I think that makes more sense than the weird list stuff that Mat did. Calling `frequencies_RR.lengths` would be a lot clearer IMO than `frequencies_RR[0]` everywhere

Reply:
Also: docstring (I know these are helper methods, but something really brief like "Puts the 3 frequency-related matrices together into a single result class" helps a ton when reading through this later and understanding why this method exists)